src/h5web/providers/hsds/api.ts

```diff
@@ -232,28 +221,31 @@ export class HsdsApi implements ProviderAPI {
   }
 
   private async resolveLink(
-    link: HDF5Link,
+    link: HsdsLink,
     path: string,
     depth: number
   ): Promise<Entity> {
-    if (!isHardLink(link)) {
+    if (link.class !== 'H5L_TYPE_HARD') {
       return {
         name: link.title,
         path,
-        kind: EntityKind.Link,
         attributes: [],
-        rawLink: link,
+        link: {
+          class: link.class === 'H5L_TYPE_SOFT' ? 'Soft' : 'External',
+          path: link.h5path,
+          file: link.file,
+        },
```

Comment:
HSDS is "link-aware", so we store a `Link` object in the entity. Note that we no longer specify the `kind` property to indicate that the entity is unresolved.